qcodes/instrument_drivers/Lakeshore/lakeshore_base.py

```diff
@@ -0,0 +1,471 @@
+                                     'temperature to equilibrate',
+                           unit='')
+        self.wait_tolerance(0.1)
+
+        self.add_parameter('wait_equilibration_time',
+                           set_cmd=None,
+                           get_cmd=None,
+                           vals=vals.Numbers(0, 100),
+                           label='Waiting equilibration time',
+                           docstring='Duration during which temperature has to '
+                                     'be within tolerance',
+                           unit='s')
+        self.wait_equilibration_time(0.5)
+
+        self.add_parameter('blocking_t',
+                           label='Setpoint value with blocking until it is '
+                                 'reached',
+                           docstring='Sets the setpoint value, and input '
+                                     'range, and waits until it is reached. '
+                                     'Added for compatibility with Loop.',
+                           vals=vals.Numbers(0, 400),
+                           get_parser=float,
+                           set_cmd=self._set_blocking_t)
+
+    def _set_blocking_t(self, temperature):
+        self.set_range_from_temperature(temperature)
+        self.setpoint(temperature)
+        self.wait_until_set_point_reached()
+
+    def set_range_from_temperature(self, temperature):
+        """
+        Sets the output range of this given heater from a given temperature.
+
+        The output range is determined by the limits given through the parameter
+        `range_limits`. The output range is used for temperatures between
+        the limits `range_limits[i-1]` and `range_limits[i]`; that is
+        `range_limits` is the upper limit for using a certain heater current.
+
+        Args:
+            temperature
+                temperature to set the range from
+
+        Returns:
+            the value of the resulting `output_range`, that is also available
+            from the `output_range` parameter itself
+        """
+        if self.range_limits.get_latest() is None:
+            raise RuntimeError('Error when calling set_range_from_temperature: '
+                               'You must specify the output range limits '
+                               'before automatically setting the range '
+                               '(e.g. inst.range_limits([0.021, 0.1, 0.2, '
+                               '1.1, 2, 4, 8]))')
+        i = bisect(self.range_limits.get_latest(), temperature)
+        # there is a `+1` because `self.RANGES` includes `'off'` as the first
+        # value.
+        self.output_range(self.INVERSE_RANGES[i+1])
+        return self.output_range()
+
+    def set_setpoint_and_range(self, temperature):
+        self.set_range_from_temperature(temperature)
+        self.setpoint(temperature)
+
+    def wait_until_set_point_reached(self,
```

Comment:
I think this method is overcomplicated. Why not do: 

```python
def while_not_at_setpoint(tolerance, settling_time): 
    
    t = time.time()
    t0 = t
    
    while t - t0 < settling_time: 
        
        t = time.time()
        if abs(temperature() - setpoint()) > tolerance: 
            t0 = t
        
        time.sleep(0.1)
```

Reply:
And whats up with all the verbose logging. Its distracting from what you are trying to accomplish

Reply:
sure, I will simplify and clean it up.